ocaml/sdk-gen/go/autogen/src/jsonrpc_client.go

```diff
@@ -0,0 +1,225 @@
+package xenapi
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"reflect"
+	"strings"
+	"time"
+)
+
+type Request struct {
+	JSONRPC string      `json:"jsonrpc"`
+	Method  string      `json:"method"`
+	Params  interface{} `json:"params,omitempty"`
+	ID      int         `json:"id"`
+}
+
+type ResponseError struct {
+	Code    int         `json:"code"`
+	Message string      `json:"message"`
+	Data    interface{} `json:"data,omitempty"`
+}
+
+type Response struct {
+	JSONRPC string         `json:"jsonrpc"`
+	Result  interface{}    `json:"result,omitempty"`
+	Error   *ResponseError `json:"error,omitempty"`
+	ID      int            `json:"id"`
+}
+
+func paramsParse(params ...interface{}) interface{} {
+	var finalParams interface{}
+	finalParams = params
+	if len(params) == 1 {
```

Comment:
Could some comments be added to explain why this is required when there is only one `parameter`?
And why is a `finalParams` required here, as it gets a copy of a slice, which means it actually refers to the same memory location of `params`? This is because I understand the `params` is a slice rather than an array.

Reply:
The paramParse is to handle different types of parameter support way in method. eg.
Examples:
```go
Examples:
	1.Nil:                                      sendCall(ctx, "method_name") -> {"method": "method_name"}
	2.Single value (simple type):  sendCall(ctx, "method_name", 1) -> {"method": "method_name", "params": [1]}
	3.Single value (slice/array):    sendCall(ctx, "method_name", []int{1, 2, 3}) -> {"method": "method_name", "params": [1, 2, 3]}
        4.Single value (struct):          sendCall(ctx, "method_name", &Person{Name: "Alex", Age: 35}) -> {"method": "method_name", "params": {"name": "Alex", "age": 35}}
	5.Multiple values:                  sendCall(ctx, "method_name", 1, 2, 3) -> {"method": "method_name", "params": [1, 2, 3]}
 ``` 
For case 3,4 ,we need to set finalParams = params[0].

Reply:
After discussion about details , we think it's actually not needed, we only need to support 1, 2,5. remove this function.